[core] Add safety and additional logging to replace debug breaks

claywar opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?
Adds some additional checks and logging around dynamic_cast, and converts some debug break macros to logging to sanely exit function.
<!-- Describe what your PR does here. If it closes an existing issue, you can mention: "Closes #1234" and GitHub will link this PR to that issue. -->

## Steps to test these changes
Nothing should change, unless additional checks catch an issue.
<!-- Clear and detailed steps to test your changes here -->

claywar committed:
[core] Add additional logging and safety around dynamic_casts

claywar committed:
[cppcheck] Convert several breaks to logging outputs, fix additional warnings

claywar committed:
[cppcheck] Additional debug break to logging conversions

WinterSolstice8 commented on src/map/party.cpp:
```diff
@@ -60,29 +60,30 @@ struct CParty::partyInfo_t
-    m_PAlliance     = nullptr;
-    m_PSyncTarget   = nullptr;
-    m_PQuaterMaster = nullptr;
+    if (PEntity != nullptr && PEntity->PParty != nullptr)
```
`if (PEntity != nullptr && PEntity->PParty != nullptr)` is throwing the CI warnings, `PEntity->PParty` should be null until it's assigned in `AddMember(PEntity);`, a second construction of PParty adding a member already in a party will throw the warning, needs an accompanying edit to the `ShowWarning`

I guess some mobs are in parties for some reason which is calling this constructor on startup which is good for this PR

zach2good commented on src/map/utils/trustutils.cpp:
```diff
@@ -408,20 +408,31 @@ namespace trustutils
 
-        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_MAIN]))->setDelay((trustData->cmbDelay * 1000) / 60);
-        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_MAIN]))->setBaseDelay((trustData->cmbDelay * 1000) / 60);
+        if (mainWeapon && rangedWeapon && ammoWeapon)
```
👍

WinterSolstice8 replied:
hmm. do trusts always have main, ranged and ammo populated with weapons? I could see this condition tripping the warning and not setting damage

zach2good replied:
They don't _really_ have weapons, but internally they're modelled like they do (for better or for worse):
```cpp
        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_MAIN]))->setDamage(static_cast<uint16>(finalDamage));
        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_RANGED]))->setDamage(static_cast<uint16>(finalDamage));
        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_AMMO]))->setDamage(static_cast<uint16>(finalDamage));

        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_MAIN]))->setDelay((trustData->cmbDelay * 1000) / 60);
        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_MAIN]))->setBaseDelay((trustData->cmbDelay * 1000) / 60);

        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_RANGED]))->setDelay((trustData->cmbDelay * 1000) / 60);
        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_RANGED]))->setBaseDelay((trustData->cmbDelay * 1000) / 60);

        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_AMMO]))->setDelay((trustData->cmbDelay * 1000) / 60);
        (dynamic_cast<CItemWeapon*>(PTrust->m_Weapons[SLOT_AMMO]))->setBaseDelay((trustData->cmbDelay * 1000) / 60);
        ```

TeoTwawki replied:
modeled the same way mobs are. basically, a placeholder object to contain dmg type, weapons dmg rating, dmg rank, and delay. When SE says things like Arciela will count as having a great weapon for the purposes of a job trait, it's because of this - there's no actual equip slots to check a great swords on, so she gets to bypass that check.

claywar committed:
[cppcheck] Additional debug break to logging conversions


Undo warning change


Fix negative check for uint


Fix incorrect condition in CParty
